File: Excel/681 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/681 Split and Align.xlsx"
input = read_excel(path, range = "A1:A5")
test  = read_excel(path, range = "C2:G12", col_names = c("1", "2", "3", "4", "5"))

result = input %>%
  mutate(group= row_number()) %>%
  separate_rows(Data, sep = ", ") %>%
  mutate(col = row_number(), .by = group) %>%
  mutate(row = row_number()) %>%
  pivot_wider(names_from = col, values_from = Data) %>%
  select(-c(group, row))

all.equal(result, test, check.attributes = FALSE) # TRUE

File: Excel/600-699/681/681 Challenge.py
import pandas as pd

path = "681 Split and Align.xlsx"
input = pd.read_excel(path, usecols="A", nrows=4)
test = pd.read_excel(path, usecols="C:G", nrows=11, names=["1", "2", "3", "4", "5"])

input["RowNumber"] = input.index + 1
input = input.assign(**{"Split": input["Data"].str.split(", ")}).explode("Split")
input["row"] = range(1, len(input) + 1)
input["col"] = input.groupby("RowNumber").cumcount() + 1
input = input.pivot(index="row", columns="col", values="Split").reset_index(drop=True)
input.columns.name = None
input.columns = test.columns


print(test.equals(input)) # True